cpp/include/kvikio/shim/cufile.hpp

```diff
@@ -62,6 +63,23 @@ class cuFileAPI {
     get_symbol(DriverSetPollMode, lib, KVIKIO_STRINGIFY(cuFileDriverSetPollMode));
     get_symbol(DriverSetMaxCacheSize, lib, KVIKIO_STRINGIFY(cuFileDriverSetMaxCacheSize));
     get_symbol(DriverSetMaxPinnedMemSize, lib, KVIKIO_STRINGIFY(cuFileDriverSetMaxPinnedMemSize));
+
+    // cuFile is supposed to open and close the driver automatically but because of a bug in
+    // CUDA 11.8, it sometimes segfault. See <https://github.com/rapidsai/kvikio/issues/159>.
+    CUfileError_t const error = DriverOpen();
+    if (error.err != CU_FILE_SUCCESS) {
+      throw std::runtime_error(std::string{"cuFile error at: "} + __FILE__ + ":" +
+                               KVIKIO_STRINGIFY(__LINE__) + ": " +
+                               cufileop_status_error(error.err));
+    }
+  }
+  ~cuFileAPI()
+  {
+    CUfileError_t const error = DriverClose();
+    if (error.err != CU_FILE_SUCCESS) {
+      std::cerr << "Unable to close GDS file driver: " << cufileop_status_error(error.err)
+                << std::endl;
```

Comment:
is this a temporary change?

Reply:
No, to avoid throwing an exception in the destructor, we print a warning instead

Reply:
Right, no exceptions in the dtor.
Posted the comment because libcudf does not do any std output, thought the same might hold for kvikio